src/test/librados_test_stub/TestClassHandler.cc

```diff
@@ -30,17 +30,30 @@ void TestClassHandler::open_class(const std::string& name,
                                   const std::string& path) {
   void *handle = dlopen(path.c_str(), RTLD_NOW);
   if (handle == NULL) {
-    derr << "Failed to load class: " << dlerror() << dendl;
+    std::cerr << "Failed to load class: " << name << " (" << path << "): "
+              << dlerror() << std::endl;
     return;
   }
-  m_class_handles.push_back(handle);
+
+  // clear any existing error
+  dlerror();
 
   // initialize
   void (*cls_init)() = reinterpret_cast<void (*)()>(
-      dlsym(handle, "__cls_init"));
-  if (cls_init) {
+    dlsym(handle, "__cls_init"));
+
+  char* error = nullptr;
+  if ((error = dlerror()) != nullptr) {
+    std::cerr << "Error locating initializer: " << error << std::endl;
+  } else if (cls_init) {
+    m_class_handles.push_back(handle);
     cls_init();
+    return;
   }
+
+  std::cerr << "Class: " << name << " (" << path << ") missing initializer"
+            << std::endl;
+  dlclose(handle);
```

Comment:
could you elaborate a little bit why dlclose() addresses the issue we are running into? i just checked all object classes. all of them have `__cls_init()` defined.

Reply:
It doesn't -- I am just fixing a potential leak.